examples/scala/src/main/scala/StandaloneExample.scala

```diff
@@ -0,0 +1,128 @@
+import glide.api.GlideClient
+import glide.api.logging.Logger
+import glide.api.models.configuration.GlideClientConfiguration
+import glide.api.models.configuration.NodeAddress
+import glide.api.models.exceptions.{ClosingException, ConnectionException, ExecAbortException, TimeoutException}
+
+import scala.concurrent.{Await, CancellationException, ExecutionContext, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+import scala.jdk.CollectionConverters.*
+import scala.jdk.FutureConverters.*
+import scala.util.{Failure, Try}
+
+object StandaloneExample {
+
+    /**
+     * Creates and returns a <code>GlideClient</code> instance.
+     *
+     * This function initializes a <code>GlideClient</code> with the provided list of nodes.
+     * The <code>nodesLis</code> may contain either only primary node or a mix of primary
+     * and replica nodes. The <code>GlideClient</code> use these nodes to connect to
+     * the Standalone setup servers.
+     *
+     * @param nodesList A list of pairs where each pair
+     *     contains a host (String) and port (Int). Defaults to [("localhost", 6379)].
+     *
+     * @return An instance of <code>Future[GlideClient]</code> connected to the specified nodes.
+     */
+    private def createClient(nodesList: List[(String, Int)] = List(("localhost", 6379))): Future[GlideClient] = {
+        // Check `GlideClientConfiguration` for additional options.
+        val config = GlideClientConfiguration.builder()
+            .addresses(nodesList.map((host, port) => NodeAddress.builder().host(host).port(port).build()).asJava)
+            // Enable this field if the servers are configured with TLS.
+            //.useTLS(true)
+            .build()
+            // This cast is required in order to pass the config to createClient because the Scala type system
+            // is unable to resolve the Lombok builder result type.
+            .asInstanceOf[GlideClientConfiguration]
+
+        GlideClient.createClient(config).asScala
+    }
+
+    /**
+     * Executes the main logic of the application, performing basic operations such as SET, GET, and
+     * PING using the provided <code>GlideClient</code>.
+     *
+     * @param client An instance of <code>GlideClient</code>.
+     */
+    private def appLogic(client: GlideClient): Future[Unit] = {
+        for {
```

Comment:
why is it inside a 'for'?

Reply:
Scala has for-comprehensions, which act as monad comprehensions (inspired by Haskell's do-notation) for writing imperative code with effects (Future is used to encode the async effect).  It's idiomatic for Scala code to be written this way, even if the keyword "for" is a bit confusing, unfortunately.